libmachine/drivers/drivermaker/maker.go

```diff
@@ -0,0 +1,29 @@
+package drivermaker
+
+import (
+	"fmt"
+
+	"github.com/docker/machine/libmachine/drivers"
+	"github.com/docker/machine/libmachine/drivers/digitalocean"
+	"github.com/docker/machine/libmachine/drivers/none"
+	"github.com/docker/machine/libmachine/drivers/virtualbox"
+)
+
+func NewDriver(driverName, hostName, artifactPath string) (drivers.Driver, error) {
+	var (
+		driver drivers.Driver
+	)
+
+	switch driverName {
+	case "virtualbox":
+		driver = virtualbox.NewDriver(hostName, artifactPath)
+	case "digitalocean":
+		driver = digitalocean.NewDriver(hostName, artifactPath)
```

Comment:
Note the lack of an error return.  There's not that much which can go wrong here.  So, all of the other drivers need to be updated to not return errors as well in their "factory" methods.